### PR TITLE
Describe defaulting and validation webhooks in Serverless docs

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Updated the architecture diagram for Serverless.
- Added a separate doc describing the defaulting and validation webhooks.
- Renumbered the existing Details docs.

**Related issue(s)**
See also #8385

### CODE REVIEW

File: docs/serverless/02-01-serverless.md [hunk 1]
```diff
@@ -12,19 +12,21 @@ Serverless relies on [Knative Serving](https://knative.dev/docs/serving/) for de
 
     >**NOTE:** Function Controller sets the Node.js 12 runtime by default.
 
-2. Function Controller (FC) detects a new Function CR.
+2. Before the Function can be saved or updated, it is verified sequentially by the [defaulting and validation webhooks](#supported-webhooks).
```
Comment (magicmatatjahu) [inline docs/serverless/02-01-serverless.md:15]: Function is verified only by validation webhook, not by defaulting. Defaulting adjusts/updates/mutates missing/optional  fields.

File: docs/serverless/02-01-serverless.md [hunk 2]
```diff
@@ -12,19 +12,21 @@ Serverless relies on [Knative Serving](https://knative.dev/docs/serving/) for de
...
+3. Function Controller (FC) detects the new, validated Function CR.
 
-4. Based on the ConfigMap, FC creates a Kubernetes Job that triggers the creation of a Function image.
+4. FC creates a ConfigMap with the Function definition.
 
-5. The Job creates a Pod with the Docker image containing the Function definition. It also pushes the image to a Docker registry.
+5. Based on the ConfigMap, FC creates a Kubernetes Job that triggers the creation of a Function image.
 
-6. FC monitors the Job status. When the image creation finishes successfully, FC creates a Service CR (KService) that points to the Pod with the image.
+6. The Job creates a Pod with the Docker image containing the Function definition. It also pushes the image to a Docker registry.
```
Comment (magicmatatjahu) [inline docs/serverless/02-01-serverless.md:23]: I didn't notice it before but the sentence `The Job creates a Pod with the Docker image containing the Function definition` indicates that job has already created an image with the definition of the Function itself and doesn't build this image. Maybe, we should change it to `The Job creates a Pod, which build Function's production Docker image` or something similar?

File: docs/serverless/03-01-supported-webhooks.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+title: Supported webhooks
+type: Details
+---
+
+A newly created or updated Function CR is verified by these two webhooks:
```
Comment (magicmatatjahu) [inline docs/serverless/03-01-supported-webhooks.md:6]: As comment above.

File: docs/serverless/03-01-supported-webhooks.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+1. **Defaulting webhook** that sets the optimal [default values](https://github.com/kyma-project/kyma/blob/master/components/function-controller/pkg/apis/serverless/v1alpha1/function_defaults.go#L15) for CPU and memory requests and limits, and adds the maximum and the minimum number of replicas, if not specified already in the Function CR.
```
Comment (magicmatatjahu) [inline docs/serverless/03-01-supported-webhooks.md:8]: I think, `optimal` word is unnecessary.

File: docs/serverless/03-01-supported-webhooks.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+2. **Validation webhook** that checks if:
+
+- Minimum values requested for CPU, memory, and replicas are not lower than the [required ones](https://github.com/kyma-project/kyma/blob/master/components/function-controller/pkg/apis/serverless/v1alpha1/function_validation.go#L19).
+- Requests are lower than or equal to limits, and that the minimum number of replicas is lower than maximum one.
```
Comment (magicmatatjahu) [inline docs/serverless/03-01-supported-webhooks.md:13]: `...minimum number of replicas is lower than or equal maximum one...`

File: docs/serverless/03-01-supported-webhooks.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+- The Function CR contains all the required parameters.
+- The format of deps, envs, labels, and the Function name ([RFC 1123](https://tools.ietf.org/html/rfc1123)) is correct.
```
Comment (magicmatatjahu) [inline docs/serverless/03-01-supported-webhooks.md:15]: https://tools.ietf.org/html/rfc1035 not `1123` :P

File: docs/serverless/02-01-serverless.md [hunk 7]
```diff
@@ -12,19 +12,21 @@ Serverless relies on [Knative Serving](https://knative.dev/docs/serving/) for de
 
     >**NOTE:** Function Controller sets the Node.js 12 runtime by default.
 
-2. Function Controller (FC) detects a new Function CR.
+2. Before the Function can be saved or updated, it is first updated and then verified by the [defaulting and validation webhooks](#details-supported-webhooks) respectively.
```
Comment (bszwarc) [inline docs/serverless/02-01-serverless.md:15]: I would maybe write `saved or modified` to distinguish one update from another.

File: docs/serverless/03-01-supported-webhooks.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
+---
+title: Supported webhooks
+type: Details
+---
+
+A newly created or updated Function CR is first updated by the defaulting webhook and then verified by the validation webhook before the Function Controller starts to process it:
```
Comment (bszwarc) [inline docs/serverless/03-01-supported-webhooks.md:6]: Same as above, maybe A newly created or modified Function CR is first updated by the...

**Pull request merged**